File: sdk/python/tempmail_sdk/providers/jama_trenet_eu.py
"""Mailinator 姊妹域名 jama.trenet.eu"""

from typing import List
from ..types import Email, EmailInfo
from . import mailinator

CHANNEL = "jama-trenet-eu"
DOMAIN = "jama.trenet.eu"


def generate_email() -> EmailInfo:
    local = mailinator._random_string(12)
    return EmailInfo(channel=CHANNEL, email=f"{local}@{DOMAIN}")


def get_emails(token: str, email: str = "", **kwargs) -> List[Email]:
    return mailinator.get_emails(token, email, **kwargs)

File: sdk/rust/src/providers/jama_trenet_eu.rs
/*!
 * Mailinator 官方姊妹域名 jama.trenet.eu
 */

use crate::providers::mailinator;
use crate::types::{Channel, Email, EmailInfo};
use rand::Rng;

const DOMAIN: &str = "jama.trenet.eu";

fn random_string(len: usize) -> String {
    const CHARS: &[u8] = b"abcdefghijklmnopqrstuvwxyz0123456789";
    let mut rng = rand::thread_rng();
    (0..len)
        .map(|_| CHARS[rng.gen_range(0..CHARS.len())] as char)
        .collect()
}

pub fn generate_email() -> Result<EmailInfo, String> {
    let local = random_string(12);
    Ok(EmailInfo {
        channel: Channel::JamaTrenetEu,
        email: format!("{}@{}", local, DOMAIN),
        token: None,
        expires_at: None,
        created_at: None,
    })
}

pub fn get_emails(email: &str) -> Result<Vec<Email>, String> {
    mailinator::get_emails(email)
}
